ray-operator/controllers/ray/utils/serve_api_models.go

```diff
@@ -0,0 +1,82 @@
+package utils
+
+type RayServeConfigType string
+
+const (
+	MULTI_APP  RayServeConfigType = "MULTI_APP"
+	SINGLE_APP RayServeConfigType = "SINGLE_APP"
+)
+
+// V1 Serve API Response format
+type ServeAppStatusInfoV1 struct {
+	Status  string `json:"status,omitempty"`
+	Message string `json:"message,omitempty"`
+}
+
+type ServeSingleApplicationStatusV1 struct {
+	ApplicationStatus  ServeAppStatusInfoV1    `json:"app_status,omitempty"`
+	DeploymentStatuses []ServeDeploymentStatus `json:"deployment_statuses,omitempty"`
+}
+
+// ServeDeploymentStatus and ServeApplicationStatus describe the format of status(es) that will
+// be returned by GetSingleApplicationStatus and GetMultiApplicationStatus methods of the dashboard client
+// Describes the status of a deployment
+type ServeDeploymentStatus struct {
+	Name    string `json:"name,omitempty"`
+	Status  string `json:"status,omitempty"`
+	Message string `json:"message,omitempty"`
+}
+
+// Describes the status of an application
+type ServeApplicationStatus struct {
+	Name        string                           `json:"name,omitempty"`
+	Status      string                           `json:"status"`
+	Message     string                           `json:"message,omitempty"`
+	Deployments map[string]ServeDeploymentStatus `json:"deployments"`
+}
+
+// V2 Serve API Response format. These extend the ServeDeploymentStatus and ServeApplicationStatus structs,
+// but contain more information such as route prefix because the V2/multi-app GET API fetchs general metadata,
+// not just statuses.
+type ServeDeploymentDetails struct {
+	ServeDeploymentStatus
+	RoutePrefix string `json:"route_prefix,omitempty"`
+}
+
+type ServeApplicationDetails struct {
+	ServeApplicationStatus
+	RoutePrefix string                            `json:"route_prefix,omitempty"`
+	DocsPath    string                            `json:"docs_path,omitempty"`
+	Deployments map[string]ServeDeploymentDetails `json:"deployments"`
+}
+
+type ServeDetails struct {
+	Applications map[string]ServeApplicationDetails `json:"applications"`
+	DeployMode   string                             `json:"deploy_mode,omitempty"`
+}
+
+// ServeConfigSpec defines the desired state of RayService, used by Ray Dashboard.
+type ServeConfigSpec struct {
```

Comment:
Can we add links to the schemas in this file?

Reply:
Added link to Serve docs for single-app schema. Will ad multi-app link when I add the multi-app schema in follow up PR